### ISSUE TITLE
Is there better documentation?

### ISSUE THREAD

**Issue by rolinger** [AuthorizeNet/sdk-php#402]
Is there better documentation hiding somewhere?

Without the few limited examples I would not have been able to get a barebones procedure written that returns me a token.  But beyond that, without having to go read the source code, where can I find real documentation that shows what each construct is supposed to be.  

For example, in the following code, that was pulled from one of the examples:
```
    //create a transaction
    $transactionRequestType = new AnetAPI\TransactionRequestType();
    $transactionRequestType->setTransactionType("authCaptureTransaction");
    $transactionRequestType->setAmount($sendData['balance']);
```

How would I know that these were the method types and the proper calls to setting the object values?  If there is no documentation (which would be on par for what I have seen related to Authorize.net in general) Is there a FULL example some where I can pull the proper usage for each of the method types?

**Comment by rickmacgillis** [comment]
Unfortunately Visa is to rich to care about making sure their products work. As long as they can fill their coffers still, they'll stop supporting tech people rely on, like this project. It's the disgusting truth about capitalism.

**Comment by gillytech** [comment]
Yeah I wouldn't count on there being any documentation other than what's [here](https://developer.authorize.net/api/reference/index.html). To get any more in-depth knowledge I have to randomly search the source code and muddle through.

**Comment by rolinger** [comment]
I have finally managed to get my first full payment transactions working and create customer profiles using that transaction.  But man, this has been VERY painful.  Its like their documentation was written by 5 different departments that have no idea what each is doing. 

The API methods are atrocious.  All the documentation is flat (left-aligned) so its really hard to figure what is a method versus what is a method parameter.  In some cases I have seen a single `*` as method, and in other cases I have seen a single `*` as a method property.  As well, lets create a customer profile ID shall we, one would think you would call::
```
      $profileType = new AnetAPI\CustomerProfileType();
      $profileType->setCustomerProfileId($ss['authCustProfileID']) ;
      $transactionRequestType->setProfile($profileType) ;
```

Uh....but nope, its actually this:
```
      $profileType = new AnetAPI\CustomerProfilePaymentType();
      $profileType->setCustomerProfileId($ss['authCustProfileID']) ;
      $transactionRequestType->setProfile($profileType) ;
```

Sure, it makes sense, lets pass the `customerProfileId` by using the `CustomerProfilePaymentType` instead - ok, that was sarcasm.

Possibly the worst documentation I have ever seen.....especially considering were talking about one of the largest, if not THE largest, merchant gateway processor in the world.  Have you call their tech support??!!!   Fuh-git-aboutit - might as well be calling them for medical advice, because it doesn't matter what the subject is.....they are pretty clueless.  I was actually told by one gal from tech support "we aren't a web developer company" and she proceeded to simply point me back to the same pathetic API docs I was calling to inquire about in the first place.   NO....your not a web development company, but the product you sell is pretty much 99% coding centric - you better have some damn coders who know the f'n system to assist.

This has been a VERY frustrating experience.  I coded FattMerchant in 3 days....Authorize.net so far has taken me 14 and I am still working on it.   Its god d**ned exhausting!

**Comment by rickmacgillis** [comment]
> I have finally managed to get my first full payment transactions working and create customer profiles using that transaction. But man, this has been VERY painful. Its like their documentation was written by 5 different departments that have no idea what each is doing.
> 
> The API methods are atrocious. All the documentation is flat (left-aligned) so its really hard to figure what is a method versus what is a method parameter. In some cases I have seen a single `*` as method, and in other cases I have seen a single `*` as a method property. As well, lets create a customer profile ID shall we, one would think you would call::
> 
> ```
>       $profileType = new AnetAPI\CustomerProfileType();
>       $profileType->setCustomerProfileId($ss['authCustProfileID']) ;
>       $transactionRequestType->setProfile($profileType) ;
> ```
> 
> Uh....but nope, its actually this:
> 
> ```
>       $profileType = new AnetAPI\CustomerProfilePaymentType();
>       $profileType->setCustomerProfileId($ss['authCustProfileID']) ;
>       $transactionRequestType->setProfile($profileType) ;
> ```
> 
> Sure, it makes sense, lets pass the `customerProfileId` by using the `CustomerProfilePaymentType` instead - ok, that was sarcasm.
> 
> Possibly the worst documentation I have ever seen.....especially considering were talking about one of the largest, if not THE largest, merchant gateway processor in the world. Have you call their tech support??!!! Fuh-git-aboutit - might as well be calling them for medical advice, because it doesn't matter what the subject is.....they are pretty clueless. I was actually told by one gal from tech support "we aren't a web developer company" and she proceeded to simply point me back to the same pathetic API docs I was calling to inquire about in the first place. NO....your not a web development company, but the product you sell is pretty much 99% coding centric - you better have some damn coders who know the f'n system to assist.
> 
> This has been a VERY frustrating experience. I coded FattMerchant in 3 days....Authorize.net so far has taken me 14 and I am still working on it. Its god d**ned exhausting!

lol That's WHY they do that. They don't care about you. They care about cash, and as long as they're at the top, they don't have a reason to grow. The best thing we can do to make them start to care is to destroy capitalism. That'll remove them from their power horse, but it's more of a decades-long process at best.
